components/request_otr/browser/request_otr_tab_storage.h

```diff
@@ -0,0 +1,70 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_REQUEST_OTR_BROWSER_REQUEST_OTR_TAB_STORAGE_H_
+#define BRAVE_COMPONENTS_REQUEST_OTR_BROWSER_REQUEST_OTR_TAB_STORAGE_H_
+
+#include "base/memory/ref_counted.h"
+#include "brave/components/brave_shields/browser/blocked_domain_1pes_lifetime.h"
+#include "content/public/browser/web_contents_user_data.h"
+
+using brave_shields::BlockedDomain1PESLifetime;
+
+namespace content {
+class WebContents;
+}  // namespace content
+
+namespace ephemeral_storage {
+class EphemeralStorageService;
+}  // namespace ephemeral_storage
+
+namespace request_otr {
+
+// A short-lived, per tab storage for Request-OTR interstitials, that stores a
+// flag while proceeding so a new interstitial is not shown immediately,
+// and stores the result of the user's request.
+class RequestOTRTabStorage
```

Comment:
This is a bit confusing to me because WebContentsUserData are normally named `*TabHelper`, but also we already have RequestOTRTabHelper. Given that it's not possible to use this outside of desktop/android, putting the code in a component may be unnecessarily complicating things. You could also call this something like `RequestOTRStorageTabHelper` or something along those lines?

Reply:
name changes can be moved to follow-up